src/shared/modules/user/user.controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Response } from 'express';
+import { StatusCodes } from 'http-status-codes';
+import { inject, injectable } from 'inversify';
+
+import { fillDTO } from '../../helpers/common.js';
+import { Config, RestSchema } from '../../libs/config/index.js';
+import { Logger } from '../../libs/logger/index.js';
+import { BaseController, HttpError } from '../../libs/rest/index.js';
+import { Component } from '../../types/component.enum.js';
+import { HttpMethod } from '../../types/index.js';
+import { CreateUserRequest } from './create-user-request.type.js';
+import { LoginUserRequest } from './login-user-request.type.js';
+import { UserRdo } from './rdo/user.rdo.js';
+import { UserService } from './user-service.interface.js';
+
+@injectable()
+export class UserController extends BaseController {
+  constructor(
+    @inject(Component.Logger) protected readonly logger: Logger,
+    @inject(Component.UserService) private readonly userService: UserService,
+    @inject(Component.Config)
+    private readonly configService: Config<RestSchema>,
+  ) {
+    super(logger);
+
+    this.logger.info('Register routes for UserController…');
+
+    this.addRoute({
+      path: '/register',
+      method: HttpMethod.Post,
+      handler: this.create,
+    });
+    this.addRoute({
+      path: '/login',
+      method: HttpMethod.Post,
+      handler: this.login,
+    });
+  }
+
+  public async create(
+    { body }: CreateUserRequest,
+    res: Response,
+  ): Promise<void> {
+    const existingUser = await this.userService.findByEmail(body.email);
+
+    if (existingUser) {
+      throw new HttpError(
+        StatusCodes.CONFLICT,
+        `User with email «${body.email}» exists.`,
+        'UserController',
+      );
+    }
+
+    const result = await this.userService.create(
+      body,
+      this.configService.get('SALT'),
+    );
+
+    this.created(res, fillDTO(UserRdo, result));
+  }
+
+  public async login({ body }: LoginUserRequest, _res: Response) {
+    const user = await this.userService.findByEmail(body.email);
+
+    if (!user) {
+      throw new HttpError(
+        StatusCodes.UNAUTHORIZED,
+        `User with email ${body.email} not found.`,
+        'UserController',
```

Comment:
Если хочешь передать названия класса, лучше этого делать используя свойство `name` у класса. Таким образом, если менять названия класса или прямо удалить его, это бросится ошибка.

```suggestion
        UserController.name,
```